### DISCUSSION TITLE
How to convert HEIC image to *.jpg with a minimum compression?

### DISCUSSION THREAD

**Discussion by tobwz** [ImageMagick/ImageMagick#7436] [Help]
### ImageMagick version

7.1.1._32

### Operating system, version and so on

Win 10

### Description

How can I  convert an HEIC image to *.jpg with a minimum compression?

**Comment by snibgo** [comment]
```
magick in.heic -quality 100 out.jpg
```

**Comment by tobwz** [comment]
Thank you.

One related additional question:

Assume I want to preserve the file timestamps (including the internal EXIF tags): How can I specify this?

`magick in.heic -define preserve-timestamp=true -quality 100 out.jpg`

does NOT work

**Reply by fmw42** [reply]
Define documentation says:

`preserve-timestamp=true\|false | Preserve file timestamp (mogrify only)`

I am not sure why it is limited to mogrify?  Sorry, I do not know something for magick.  A developer might comment further.

**Comment by snibgo** [comment]
`-define preserve-timestamp=true` is limited to `mogrify`. (I don't know why.) It preserves the file timestamps, as seen by the operating system (creation, last_access and last_write). It has no effect on any timestamp metadata within the file.

exiftool can be used to copy internal metadata from one file to another.

**Comment by tobwz** [comment]
If I use mogrify instead of magick I am getting errors like:

mogrify: no encode delegate for this image format `HEIC' @ warning/constitute.c/WriteImage/1400.
mogrify: no encode delegate for this image format `HEIC' @ error/constitute.c/WriteImage/1409.
mogrify: unable to open image 'D:\immag\test.jpg': No such file or directory @ error/blob.c/OpenBlob/3596.

**Comment by snibgo** [comment]
> no encode delegate for this image format HEIC

You are trying to write to ("encode") HEIC format. `magick -list format` will tell you that IM can't write to HEIC.

Please show your command.

**Comment by tobwz** [comment]
Ok, here is my command:

```
mogrify test.heic -define preserve-timestamp=true -quality 100 out.jpg
mogrify: no encode delegate for this image format `HEIC' @ warning/constitute.c/WriteImage/1400.
mogrify: no encode delegate for this image format `HEIC' @ error/constitute.c/WriteImage/1409.
mogrify: unable to open image 'out.jpg': No such file or directory @ error/blob.c/OpenBlob/3596.
```

**Comment by snibgo** [comment]
If you want `-define preserve-timestamp=true` then you need mogrify. But mogrify is for multiple inputs, and makes each output the same filname as the correspolnding input. I suggest:
```
magick mogrify -format jpg -define preserve-timestamp=true -quality 100 test.heic
```
This will make the file `test.jpg`. Then use an OS command to rename it to `out.jpg`, if you want.
